### DISCUSSION TITLE
GraphQL: Getting checkbox/dropdown labels within tables

### DISCUSSION THREAD

**Discussion by gregorterrill** [craftcms/cms#8814] [Ideas]
GraphQL support for the label argument was added to Checkbox, Dropdown etc. fields here: #5514 but this doesn't seem to extend to checkboxes, dropdowns etc. inside Table field rows. Is it possible for this to be extended so we can grab the labels from fields within Table fields as well?

**Comment by brandonkelly** [comment]
It’s not even possible to access Table dropdown option labels within Twig, and not sure it’s worth the effort. But I’ll leave this open.

**Reply by alinnert** [reply]
I just ran into a situation where I need access to the label of a dropdown in a table. But maybe there's another way to solve what I'm trying to do. Here's what I want to achieve:

On a website there's a table for a weekly plan. Something like this:

| Day of week | Time  | Event   |
| ----------- | ----- | ------- |
| Monday      | 10:00 | Event A |
|             | 14:00 | Event B |
| Tuesday     | 10:00 | Event C |

I thought a table would be perfectly suitable for this. Now I use a dropdown for the day of week with the following settings:

| Value | Label   |
| ----- | ------- |
| 0     | Monday  |
| 1     | Tuesday |
| etc.  |         |

I could repeat the label in the value field but I already use the numbers for sorting purposes. Everything besides a table feels so overkill for this use-case. What I can do is using something like `0|Monday` as the value and extract the label that way. That works, it just feels a bit weird and requires me to repeat or copy and paste all labels.

**Reply by brandonkelly** [reply]
Since it’s not possible for authors to define custom dropdown options, it should be save to hard code each of the option labels, mapped to their values.
